pkg/api/resource/helpers.go

```diff
@@ -28,47 +28,41 @@ import (
 // PodRequestsAndLimits returns a dictionary of all defined resources summed up for all
 // containers of the pod.
 func PodRequestsAndLimits(pod *api.Pod) (reqs map[api.ResourceName]resource.Quantity, limits map[api.ResourceName]resource.Quantity) {
-	reqs, limits = map[api.ResourceName]resource.Quantity{}, map[api.ResourceName]resource.Quantity{}
-	for _, container := range pod.Spec.Containers {
-		for name, quantity := range container.Resources.Requests {
-			if value, ok := reqs[name]; !ok {
-				reqs[name] = *quantity.Copy()
-			} else {
-				value.Add(quantity)
-				reqs[name] = value
-			}
-		}
-		for name, quantity := range container.Resources.Limits {
-			if value, ok := limits[name]; !ok {
-				limits[name] = *quantity.Copy()
+	add := func(list map[api.ResourceName]resource.Quantity, containerResList api.ResourceList) map[api.ResourceName]resource.Quantity {
+		for name, quantity := range containerResList {
+			if value, ok := list[name]; !ok {
+				list[name] = *quantity.Copy()
 			} else {
 				value.Add(quantity)
-				limits[name] = value
+				list[name] = value
```

Comment:
if this is mutating `list` in place, why do we also need to return it? that's makes this a little hard to follow.

also, could `add` and `max` be separate private functions? making them closures makes it hard to follow if they depend on variables from the PodRequestsAndLimits scope

Reply:
The function do be better.  Done. Thanks